core/src/wasm/overrides.rs

```diff
@@ -0,0 +1,340 @@
+            if let Ok(id) = js_sys::Reflect::get(&event.data(), &JsValue::from_str("id")) {
+                // Get the callback function from the map
+                let callback = callbacks_clone.get(&id);
+                // Check if the callback is defined
+                let undefined = callback.is_undefined();
+                if !undefined {
+                    // If the callback is defined, call it
+                    if let Some(func) = callback.dyn_ref::<js_sys::Function>() {
+                        func.call0(&JsValue::NULL).unwrap();
+                    }
+                }
+            }
+        }) as Box<dyn FnMut(_)>); // As a Box<dyn FnMut(_)> type which is a function that takes a single argument
+
+        // Set the closure as the onmessage handler for the worker
+        worker_clone.set_onmessage(Some(closure.as_ref().unchecked_ref()));
+
+        // Forget the closure so it is not deallocated from memory
+        closure.forget();
+
+        TimerManager {
+            worker,
+            next_id: 1,
+            callbacks,
+        }
+    }
+
+    /// Replacement for setInterval
+    #[wasm_bindgen(js_name = "setInterval")]
+    pub fn set_interval(&mut self, callback: &JsValue, time: u32) -> u32 {
+        // Get the current ID and increment it
+        let id = self.next_id;
+        self.next_id += 1;
+
+        // Store the callback in the map with the ID as the key
+        self.callbacks.set(&JsValue::from(id), callback);
+
+        let request = js_sys::Object::new();
+        // Set the name of the function to call which is setInterval
+        js_sys::Reflect::set(
+            &request,
+            &JsValue::from("name"),
+            &JsValue::from("setInterval"),
+        )
+        .unwrap();
+
+        // Set the ID of the interval
+        js_sys::Reflect::set(&request, &JsValue::from("id"), &JsValue::from(id)).unwrap();
+
+        // Set the time of the interval
+        js_sys::Reflect::set(&request, &JsValue::from("time"), &JsValue::from(time)).unwrap();
+
+        // Post the message to the worker
+        self.worker.post_message(&request).unwrap();
+
+        // Return the ID of the interval
+        id
+    }
+
+    #[wasm_bindgen(js_name = "clearInterval")]
+    pub fn clear_interval(&mut self, id: u32) {
+        let request = js_sys::Object::new();
+
+        // Set the name of the function to call which is clearInterval
+        js_sys::Reflect::set(
+            &request,
+            &JsValue::from("name"),
+            &JsValue::from("clearInterval"),
+        )
+        .unwrap();
+
+        // Set the ID of the interval
+        js_sys::Reflect::set(&request, &JsValue::from("id"), &JsValue::from(id)).unwrap();
+
+        // Post the message to the worker
+        self.worker.post_message(&request).unwrap();
+
+        // Delete the callback from the map
+        self.callbacks.delete(&JsValue::from(id));
+    }
+
+    #[wasm_bindgen(js_name = "setTimeout")]
```

Comment:
As I understand there is no need to export `setTimeout` and other functions

Reply:
a2f25c0efe6a1120867a044ce3e95715ed5d9804